include/download.mk

```diff
@@ -214,14 +214,15 @@ endef
 
 # Only intends to be called as a submethod from other DownloadMethod
 define DownloadMethod/rawgit
-	echo "Checking out files from the git repository..."; \
+	gitdir="$(DL_DIR)/git/`basename $(URL)`"; \
+	echo "Checking out files to $$$$gitdir ..."; \
 	mkdir -p $(TMP_DIR)/dl && \
 	cd $(TMP_DIR)/dl && \
 	rm -rf $(SUBDIR) && \
 	[ \! -d $(SUBDIR) ] && \
-	git clone $(OPTS) $(URL) $(SUBDIR) && \
-	(cd $(SUBDIR) && git checkout $(VERSION) && \
-	git submodule update --init --recursive) && \
+	mkdir -p "$(SUBDIR)" && \
+	mkdir -p $(DL_DIR)/git && \
+	"$(SCRIPT_DIR)"/git-cached-checkout.sh "$$$$gitdir" "$(URL)" "$(VERSION)" "$(TMP_DIR)/dl/$(SUBDIR)" && \
```

Comment:
It shouldn't be enabled by default, it should be a development config option.

Reply:
I would try to keep it simple, so probably define `define DownloadMethod/rawgit-cached` and use that if some development config option is enabled.

Reply:
> It shouldn't be enabled by default, it should be a development config option.

I'd suggest it is enabled always. No need for configurable since the "dl" folder is a cache in the first place. And since @Ansuel pointed out, git isn't used if package is available for download from openwrt servers, git (either cached or not) is only used for development or non-official feeds